src/debug.h

```diff
@@ -53,6 +53,12 @@
 #define __FUNCTION_NAME__ __func__
 #endif
 
+#define fatal(cond) \
+    if( !( cond ) ) { \
+        debugmsg( "Assertion failed: %s", STRING( cond ) ); \
+        std::terminate(); \
```

Comment:
I understand, but it's a regression.

A simpler fix, if you can deal with assert not meaning quite what you think it should mean, is to leave assert() alone and have something like:

```
#ifdef RELEASE
#  define debugassert(x)
#else
#  define debugassert(x) assert(x)
#endif
```
which retains all the existing platform-specific parts of assert.